src/data/finsupp/basic.lean

```diff
@@ -774,8 +775,75 @@ end
 @[simp] lemma graph_eq_empty {f : α →₀ M} : f.graph = ∅ ↔ f = 0 :=
 (graph_injective α M).eq_iff' graph_zero
 
+/-- When turned into a list of key/value pairs, a graph has no duplicate keys. -/
+lemma graph_nodupkeys (f : α →₀ M) : (list.map prod.to_sigma f.graph.to_list).nodupkeys :=
+begin
+  rw [list.nodupkeys, list.keys, list.map_map, prod.fst_to_sigma_comp, list.nodup_map_iff_inj_on],
+  { rintros ⟨b, m⟩ hb ⟨c, n⟩ hc (rfl : b = c),
+    rw [mem_to_list, finsupp.mem_graph_iff] at hb hc,
+    dsimp at hb hc,
+    rw [←hc.1, hb.1] },
+  { apply nodup_to_list }
+end
+
 end graph
 
+end finsupp
+
+/-! ### Declarations about `list.to_finsupp` and `alist.to_finsupp` -/
+
+namespace list
+variable [has_zero M]
+
+/-- Converts a list of key/value pairs into a finitely supported function via `list.lookup`. -/
+def to_finsupp (l : list (sigma (λ x : α, M))) : α →₀ M :=
+finsupp.of_support_finite (λ a, (l.lookup a).get_or_else 0) begin
+  apply l.keys.to_finset.finite_to_set.subset,
```

Comment:
Can you not construct the support explicitly?

Reply:
Yes, and I've golfed accordingly.